.github/workflows/code_coverage.yml

```
@@ -0,0 +1,59 @@
+name: Code coverage
+
+on:
+  workflow_run:
+    workflows: ["Integration tests"]
+    types: [completed]
+    branches: ['**']
+
+jobs:
+  run-codecov:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Download unit tests artifacts
+        uses: dawidd6/action-download-artifact@v2
+        with:
+          workflow: unit_tests.yml
+          name: artifacts_ut
+          path: artifacts_ut
+          pr: ${{github.event.pull_request.number}}
+      - name: Download functional tests artifacts
+        uses: dawidd6/action-download-artifact@v2
+        with:
+          workflow: functional_tests.yml
+          name: artifacts_ft
+          path: artifacts_ft
+          pr: ${{github.event.pull_request.number}}
+      - name: Download integration tests artifacts
+        uses: dawidd6/action-download-artifact@v2
+        with:
+          workflow: integration_tests.yml
+          name: artifacts_int
+          path: artifacts_int
+          pr: ${{github.event.pull_request.number}}
+      - name: Upload global coverage to Codecov
+        uses: codecov/codecov-action@v1
+        with:
+          files: ./artifacts_ut/coverage_linux.xml,./artifacts_ut/coverage_macos.xml,./artifacts_ut/coverage_windows.xml,./artifacts_ft/coverage_linux.xml,./artifacts_ft/coverage_macos.xml,./artifacts_ft/coverage_windows.xml,./artifacts_int/coverage_windows.xml
+          flags: global
+          env_vars: PYTHON
```

```suggestion
          env_vars: OS,PYTHON
```
Do that for all othe roccurrences.